.github/workflows/receive-pr.yml

```diff
@@ -0,0 +1,54 @@
+# Description: This workflow runs OpenRewrite recipes against opened pull request and upload the patch.
+# Since this pull request receives untrusted code, we should **NOT** have any secrets in the environment.
+# https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+---
+name: receive-pr
+
+on:
+  pull_request:
+    types: [opened, synchronize]
```

Comment:
Runs are quick; around ~1 minute locally here, so I figured no harm in doing this for every sync of a PR. Can be changed to just `opened` if you want to limit the runs.